src/bb8_backup.py

```diff
@@ -0,0 +1,43 @@
+from subprocess import run
+
+targets = ["orderly"]
+finished_setup = False
+bb8_dir = "../montagu-bb8/bb8"
+
+
+def requires_bb8_setup(func):
+    def decorated(*args, **kwargs):
+        global finished_setup
+        if not finished_setup:
+            setup()
+        return func(*args, **kwargs)
+
+    return decorated
+
+
+def setup():
+    global finished_setup
+    template = "- Configuring and installing bb8 backup service with these " \
+               "targets: {}"
+    print(template.format(targets))
+    args = ["./setup.sh", "../config.json"] + targets
+    run(args, cwd=bb8_dir, check=True)
+    finished_setup = True
+
+
+@requires_bb8_setup
+def backup():
+    print("Performing bb8 backup")
+    run(["bb8", "backup"], check=True)
+
+
+@requires_bb8_setup
+def schedule():
+    print("Scheduling bb8 backup")
+    run("./schedule", cwd=bb8_dir, check=True)
+
+
+@requires_bb8_setup
+def restore():
```

Comment:
This no longer needs `service` (vs the duplicati version) which means that the volume names are duplicated between the docker-compose.yml and the montagu-bb8 configuration https://github.com/vimc/montagu-bb8/blob/master/config.json

Can we (at some point, not necessarily here) move to generating the json used by bb8 from the configuration as used by montagu?

Reply:
That sounds like a good idea: https://vimc.myjetbrains.com/youtrack/issue/VIMC-1693